assets/js/modules/analytics/datastore/settings.test.js

```diff
@@ -333,6 +339,70 @@ describe( 'modules/analytics settings', () => {
 
 				expect( ( await getItem( cacheKey ) ).value ).toBeFalsy();
 			} );
+
+			describe( 'analytics-4', () => {
+				beforeEach( () => {
+					registry.dispatch( STORE_NAME ).receiveGetExistingTag( null );
+					registry.dispatch( STORE_NAME ).setSettings( validSettings );
+
+					enabledFeatures.add( 'ga4setup' );
+				} );
+
+				afterEach( () => {
+					enabledFeatures.delete( 'ga4setup' );
+				} );
+
+				it( 'should save analytics-4 settings as well', async () => {
+					const ga4Settings = {
+						propertyID: '1000',
+						webDataStreamID: '2000',
+					};
+
+					fetchMock.postOnce( gaSettingsEndpoint, { body: validSettings, status: 200 } );
+					fetchMock.postOnce( ga4SettingsEndpoint, { body: ga4Settings, status: 200 } );
+
+					registry.dispatch( MODULES_ANALYTICS_4 ).setSettings( ga4Settings );
+
+					expect( registry.select( STORE_NAME ).haveSettingsChanged() ).toBe( true );
+					expect( registry.select( MODULES_ANALYTICS_4 ).haveSettingsChanged() ).toBe( true );
+
+					const { error: saveChangesError } = await registry.dispatch( STORE_NAME ).submitChanges();
+					expect( saveChangesError ).toBeUndefined();
+
+					expect( fetchMock ).toHaveFetched( gaSettingsEndpoint, { body: { data: validSettings } } );
+					expect( fetchMock ).toHaveFetched( ga4SettingsEndpoint, { body: { data: ga4Settings } } );
+
+					expect( registry.select( STORE_NAME ).haveSettingsChanged() ).toBe( false );
+					expect( registry.select( MODULES_ANALYTICS_4 ).haveSettingsChanged() ).toBe( false );
+				} );
+
+				it( 'should ignore analytics-4 errors if it fails', async () => {
```

Comment:
I'm not sure if this is the expected behavior or not as it isn't in the ACs, but I would expect an error from GA4 to be returned from UA `submitChanges` if there was one since it's functioning as a combined method.

@felixarntz what should happen in the event of a GA4 error when submitting changes?

Reply:
@aaemnnosttv Indeed it looks like that's missing in the ACs, but it is correct (and mentioned in the ACs of the related #3244). The reason that errors here should be ignored from a user perspective is that the APIs powering these interactions are unstable, and if something fails with GA4 for that reason, it's not a blocker from proceeding.

This should all change once the GA4 APIs are stable. For now, we only should track such errors to have awareness if we need to make changes (e.g. in a hotfix), which should already be the case through the regular API error tracking for opted-in users (correct @eugene-manuilov?).

Reply:
Yep, we track it here: https://github.com/google/site-kit-wp/blob/db4d6444f12e4185f1ff36e2602e3b6fe14a1fc6/assets/js/googlesitekit/api/index.js#L171-L173